packages/perseus/src/widgets/interactive-graphs/interactive-graph-validator.test.ts

```diff
@@ -1,17 +1,66 @@
 import invariant from "tiny-invariant";
 
 import {clone} from "../../../../../testing/object-utils";
+import {allGraphCorrectKeys} from "../../perseus-types";
 
 import interactiveGraphValidator from "./interactive-graph-validator";
 
-import type {PerseusGraphType} from "../../perseus-types";
+import type {
+    PerseusGraphType,
+    PerseusGraphCorrectType,
+} from "../../perseus-types";
 import type {PerseusInteractiveGraphRubric} from "../../validation.types";
 import type {Coord} from "@khanacademy/perseus";
 
 function createRubric(graph: PerseusGraphType): PerseusInteractiveGraphRubric {
-    return {graph, correct: graph};
+    const rubricCorrect = {};
+    for (const key of Object.keys(graph) as Array<keyof typeof graph>) {
+        if (allGraphCorrectKeys[graph["type"]].includes(key)) {
+            rubricCorrect[key] = graph[key];
+        }
+    }
+    return {graph, correct: rubricCorrect as PerseusGraphCorrectType};
```

Comment:
I think there's probably a more future-proof way to do this. In order for this code to continue to work correctly, we will need to remember to update `allGraphCorrectKeys` whenever we add a new field to one of the `*GraphCorrect` types. If we forget to update `allGraphCorrectKeys`, nothing will fail, but the `correct` return value will not contain all the keys it's supposed to.

In addition, I don't think there's anything to ensure the keys listed in `allGraphCorrectKeys` are actually keys of the corresponding `*GraphCorrect` type, nor that those keys also exist (with the same value types) on `PerseusGraphType`. I.e. all these types/values could get out of sync, and we wouldn't necessarily get any type errors forcing us to fix the problem.

One way to solve this is to switch on the graph type and construct the `correct` value in a more explicit way:

```ts
function createRubric(graph: PerseusGraphType): PerseusInteractiveGraphRubric {
  switch (graph.type) {
    case "circle":
      const {type, center, radius} = graph;
      return {graph, correct: {type, center, radius}};
    case "linear":
      const {type, coords} = graph;
      return {graph, correct: {type, coords}};
    // ...
    default:
      throw new UnreachableCaseError(graph);
  }
}
```

This seems like more code, but it's typesafe and future-proof. We can get rid of the `*GraphCorrectKeys` consts, too, so it might end up being around the same amount of code overall.

Reply:
Bonus: if the first thing you do is add the `default` branch with `throw new UnreachableCaseError(graph);` I think you can just fix the TS errors one by one and you'll end up with correct code.

Reply:
Just for reference, decided not to use the helper function and instead just created the rubric object for each test.